client/js/components/map/map/DpOlMapLayer.vue

```diff
@@ -56,19 +55,44 @@ export default {
     }
   },
 
+  data () {
+    return {
+      source: null
+    }
+  },
+
   computed: {
+    defaultAttributions () {
+      const currentYear = formatDate(new Date(), 'YYYY')
+      // If a value is currently given, replace the {year} placeholder within that value
+      if (this?.attributions) {
+        return this.attributions.replace('{year}', currentYear)
+      }
+      // If not, default to the default message
+      return Translator.trans('map.attribution.default', {
+        linkImprint: Routing.generate('DemosPlan_misccontent_static_imprint'),
+        currentYear
+      })
+    },
+
     map () {
       return this.olMapState.map
     }
   },
 
+  watch: {
+    defaultAttributions () {
+      this.source.setAttributions(this.defaultAttributions)
+    }
+  },
+
   methods: {
     addLayer () {
       if (this.map === null) {
         return
       }
 
-      const source = createSourceTileWMS(this.url, this.layers, this.projection, this.attributions, this.map)
+      const source = this.source = createSourceTileWMS(this.url, this.layers, this.projection, this.defaultAttributions, this.map)
```

Comment:
when You add `this.soruce` here, I suggest to drop `source` and pass `this.source` to the method instead